diff --git a/tools/gyp/pylib/gyp/generator/ninja.py b/tools/gyp/pylib/gyp/generator/ninja.py
index 3db3771ac9..a4af2ab9ac 100644
--- a/tools/gyp/pylib/gyp/generator/ninja.py
+++ b/tools/gyp/pylib/gyp/generator/ninja.py
@@ -2845,7 +2845,7 @@ def GenerateOutputForConfig(target_list, target_dicts, data, params, config_name
             if name != target.FinalOutput() and spec["toolset"] == "target":
                 target_short_names.setdefault(name, []).append(target)
             target_outputs[qualified_target] = target
-            if qualified_target in all_targets:
+            if qualified_target in all_targets and (flavor != "ios" or "executable" != target.type):
                 all_outputs.add(target.FinalOutput())
             non_empty_target_names.add(name)
         else:
